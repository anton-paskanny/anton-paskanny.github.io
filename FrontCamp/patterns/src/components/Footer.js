export default class Footer {
  constructor() {
    // Singletone pattern
    if (!Footer.instance) {
      this.domNode = this.createElement()
      Footer.instance = this;
    }

    return Footer.instance;
  }
  createElement() {
    return document.createElement('footer');
  }
  build() {
    this.domNode.className = 'footer';
    this.domNode.innerHTML = '<p>Powered by <a href="https://newsapi.org/" title="NewsAPI">NewsAPI.org</a></p>';

    return this;
  }
  getDomNode() {
    return this.domNode;
  }
}
