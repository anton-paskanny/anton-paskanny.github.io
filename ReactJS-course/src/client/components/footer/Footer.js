import React from 'React';

export default () => (
    <footer>
        <h3>netflixroulette</h3>
    </footer>
)