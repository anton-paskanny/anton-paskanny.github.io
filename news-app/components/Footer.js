class Footer {
  constructor() {
    this.footer = this.createComponent();
  }
  createComponent() {
    return document.createElement('footer');
  }
  build() {
    this.footer.className = 'footer';
    this.footer.innerHTML = '<p>Powered by <a href="https://newsapi.org/" title="NewsAPI">NewsAPI.org</a></p>';

    return this;
  }
  getComponent() {
    return this.footer;
  }
}
