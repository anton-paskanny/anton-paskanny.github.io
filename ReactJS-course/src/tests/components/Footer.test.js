import React from 'react';
import { shallow } from 'enzyme';

import Footer from '../../client/components/Footer/Footer';

describe('Footer', () => {
    it('render component correctly', () => {
        const component = shallow(<Footer />);
        expect(component).toMatchSnapshot();
    });
});



