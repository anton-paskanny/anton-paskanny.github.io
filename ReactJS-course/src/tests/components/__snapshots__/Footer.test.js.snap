// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render component correctly 1`] = `
<footer
  className="main-footer"
>
  <p
    className="main-footer__title common-title"
  >
    netflixroulette
  </p>
</footer>
`;
